orttraining/orttraining/test/python/orttraining_run_bert_pretrain.py

```diff
@@ -524,41 +585,41 @@ def setUp(self):
-        # train_batch_size is initially configured as per optimization batch size,
-        # taking into consideration of world_size and gradient_accumulation_steps:
-        # train_batch_size = world_size * gradient_accumulation_steps * batch_size_per_gpu
-        # in the code later train_batch_size is translated to per gpu batch size:
-        # args.train_batch_size = args.train_batch_size // args.gradient_accumulation_steps // args.world_size
-
-        # the LAMB batch size of 64k
-        optimization_batch_size = 64 * 1024
-        per_gpu_batch_size = 32
-
-        self.train_batch_size = optimization_batch_size
-        self.gradient_accumulation_steps = optimization_batch_size // per_gpu_batch_size // self.world_size
+    def test_pretrain_throughput(self, process_args=None):
+        if process_args.sequence_length == 128:
+            input_dir = '/bert_data/hdf5_lower_case_1_seq_len_128_max_pred_20_masked_lm_prob_0.15_random_seed_12345_dupe_factor_5/books_wiki_en_corpus/train'
+        else:
+            input_dir = '/bert_data/hdf5_lower_case_1_seq_len_512_max_pred_80_masked_lm_prob_0.15_random_seed_12345_dupe_factor_5/books_wiki_en_corpus/train'
 
-        logger.info("self.gradient_accumulation_steps = %d", self.gradient_accumulation_steps)
+        print("process_args.enable_mixed_precision: ", process_args.enable_mixed_precision)
+        print("process_args.sequence_length: ", process_args.sequence_length)
+        print("process_args.max_batch_size: ", process_args.max_batch_size)
+        print("process_args.max_predictions_per_seq: ", process_args.max_predictions_per_seq)
+        print("process_args.gelu_recompute: ", process_args.gelu_recompute)
+        print("process_args.attn_dropout_recompute: ", process_args.attn_dropout_recompute)
+        print("process_args.transformer_layer_recompute: ", process_args.transformer_layer_recompute)
 
-        # only to run on  optimization step because we only want to make sure there is no throughput regression
-        self.max_steps = 1
         args = PretrainArguments(
-            output_dir=self.output_dir,
-            bert_model=self.bert_model,
+            input_dir=input_dir,
+            output_dir='/bert_data/hf_data/test_out/bert_pretrain_results',
+            bert_model='bert-large-uncased',
             local_rank=self.local_rank,
             world_rank=self.world_rank,
             world_size=self.world_size,
-            max_steps=self.max_steps,
-            learning_rate=self.learning_rate,
-            max_seq_length=self.max_seq_length,
-            max_predictions_per_seq=self.max_predictions_per_seq,
-            train_batch_size=self.train_batch_size,
-            gradient_accumulation_steps=self.gradient_accumulation_steps,
-            input_dir=self.input_dir,
-            fp16=self.fp16,
-            allreduce_post_accumulation=self.allreduce_post_accumulation)
+            max_steps=10,
+            learning_rate=5e-4,
```

Comment:
should these be hard-coded?

Reply:
As a nature of unit test (I did not fully agree though), it is not recommended to pass arguments as test parameters. Thus it need to be hard coded.